packages/@aws-cdk/aws-vpcv2-alpha/lib/subnet-v2.ts

```diff
@@ -0,0 +1,381 @@
+ */
+function storeSubnetToVpcByType(vpc: IVpcV2, subnet: SubnetV2, type: SubnetType) {
+  const findFunctionType = subnetTypeMap[type];
+  if (findFunctionType) {
+    findFunctionType(vpc, subnet);
+  } else {
+    throw new Error(`Unsupported subnet type: ${type}`);
+  }
+
+  /**
+   * Need to set explicit dependency as during stack deletion,
+   * the cidr blocks may get deleted first and will fail as the subnets are still using the cidr blocks
+   */
+  for (const cidr of vpc.secondaryCidrBlock) {
+    subnet.node.addDependency(cidr);
+  }
+}
+
+const subnetTypeMap = {
+  [SubnetType.PRIVATE_ISOLATED]: (vpc: IVpcV2, subnet: SubnetV2) => vpc.isolatedSubnets.push(subnet),
+  [SubnetType.PUBLIC]: (vpc: IVpcV2, subnet: SubnetV2) => vpc.publicSubnets.push(subnet),
+  [SubnetType.PRIVATE_WITH_EGRESS]: (vpc: IVpcV2, subnet: SubnetV2) => vpc.privateSubnets.push(subnet),
+  [SubnetType.ISOLATED]: (vpc: IVpcV2, subnet: SubnetV2) => vpc.isolatedSubnets.push(subnet),
+  [SubnetType.PRIVATE]: (vpc: IVpcV2, subnet: SubnetV2) => vpc.privateSubnets.push(subnet),
+  [SubnetType.PRIVATE_WITH_NAT]: (vpc: IVpcV2, subnet: SubnetV2) => vpc.privateSubnets.push(subnet),
+};
+
+/**
+ * Validates whether the provided VPC supports IPv6 addresses.
+ *
+ * @param vpc The VPC instance to be validated.
+ * @throws Error if the VPC does not support IPv6 addresses.
+ * @returns True if the VPC supports IPv6 addresses, false otherwise.
+ */
+function validateSupportIpv6(vpc: IVpcV2) {
+  if (vpc.secondaryCidrBlock.some((secondaryAddress) => secondaryAddress.amazonProvidedIpv6CidrBlock === true ||
+  secondaryAddress.ipv6IpamPoolId != undefined)) {
+    return true;
+  } else {
+    throw new Error('To use IPv6, the VPC must enable IPv6 support.');
+  }
+}
+
+/**
+ * Checks if the provided CIDR range falls within the IP address ranges of the given VPC.
+ *
+ * @param vpc The VPC instance to check against.
+ * @param cidrRange The CIDR range to be checked.
+ * @returns True if the CIDR range falls within the VPC's IP address ranges, false otherwise.
+ */
+
+function checkCidrRanges(vpc: IVpcV2, cidrRange: string) {
+
+  const vpcCidrBlock = [vpc.ipv4CidrBlock];
+
+  for (const ipAddress of vpc.secondaryCidrBlock) {
+    if (ipAddress.cidrBlock) {
+      vpcCidrBlock.push(ipAddress.cidrBlock);
+    }
+  }
+  const cidrs = vpcCidrBlock.map(cidr => new CidrBlock(cidr));
+
+  const subnetCidrBlock = new CidrBlock(cidrRange);
+
+  return cidrs.some(c => c.containsCidr(subnetCidrBlock));
+
+}
+
+/**
+ * Validates if the provided IPv4 CIDR block overlaps with existing subnet CIDR blocks within the given VPC.
+ *
+ * @param vpc The VPC instance to check against.
+ * @param ipv4CidrBlock The IPv4 CIDR block to be validated.
+ * @returns True if the IPv4 CIDR block overlaps with existing subnet CIDR blocks, false otherwise.
+ */
+
+function validateOverlappingCidrRanges(vpc: IVpcV2, ipv4CidrBlock: string): boolean {
```

Comment:
yes it is meant to be internal, without having it in export, can someone use it? i will look into @internal marking and add it here.

Reply:
may be I am very cautious because I am not sure how will JSII and the tools used to generate docs handle these functions. From the docs, it looks like we do not include any module functions level, also I believe JSII will ignore them as well, but let's double check.